src/script/lua_api/l_mainmenu.cpp

```diff
@@ -637,17 +637,33 @@ int ModApiMainMenu::l_create_world(lua_State *L)
 	const char *name	= luaL_checkstring(L, 1);
 	int gameidx			= luaL_checkinteger(L,2) -1;
 
+	std::string new_dir = fs::SanitizeFilename(name);
 	std::string path = porting::path_user + DIR_DELIM
-			"worlds" + DIR_DELIM
-			+ name;
+			+ "worlds" + DIR_DELIM + new_dir;
+
+	if (fs::PathExists(path)) {
+		path = path + "_";
+		for (int i = 1; i <= 10000; i++) {
+			std::ostringstream test_path;
+			test_path << path << i;
+			if (!fs::PathExists(test_path.str())) {
+				path = test_path.str();
+				break;
+			}
+			if (i == 10000) {
```

Comment:
due to the `i <= 10000` condition i will end up at 10001 if it fails

generally using a boolean `success` might be better to validate this